src/core/DataValidator/ConfigValidator.js

```diff
@@ -0,0 +1,239 @@
+  if (withNullType === false) {
+    const index = description.indexOf("null?");
+    if (index >= 0) description.splice(index, 1);
+  }
+
+  return description;
+};
+
+/**
+ * Flatten config tree for faster iterations and searches
+ * @param {object} tree
+ * @param {string} parent
+ * @returns {object[]}
+ */
+const flattenTree = (tree, parent = null) => {
+  const result = [];
+
+  for (let child of tree.children) {
+    /* Create a child without children and
+    assign id of the parent for quick mathcing */
+    const flatChild = { ...child, parent: parent?.id ?? null };
+    delete flatChild.children;
+
+    result.push(flatChild);
+
+    /* Recursively add children if exist */
+    if (child.children instanceof Array) {
+      result.push(...flattenTree(child, child));
+    }
+  }
+
+  return result;
+};
+
+/**
+ * Validates presence and format of the name attribute
+ * @param {Object} child
+ * @param {Object} model
+ */
+const validateNameTag = (child, model) => {
+  const { name } = model.properties;
+
+  if (name && child.name === undefined) {
+    return errorBuilder.required(model.name, "name");
+  }
+
+  return null;
+};
+
+/**
+ * Validates toName attribute
+ * Checks that connected tag is existing tag, it present in the tree
+ * and can be controlled by current Object Tag
+ * @param {Object} element
+ * @param {Object} model
+ * @param {Object[]} flatTree
+ */
+const validateToNameTag = (element, model, flatTree) => {
+  const { toname, controlledTags } = model.properties;
+
+  if (toname && element.toname === undefined) {
+    return errorBuilder.required(model.name, "toname");
+  }
+
+  if (!controlledTags) return null;
+
+  // Collect available tag names
+  const controlledTagTypes = getTypeDescription(controlledTags, false);
+
+  // Check if controlled tags are actually registered
+  const notRegisteredTags = controlledTagTypes.reduce((res, value) => {
+    try {
+      Registry.getModelByTag(value.toLowerCase());
+    } catch {
+      res.push(errorBuilder.unknownTag(model.name, "toname", value));
+    }
+    return res;
+  }, []);
+
+  if (notRegisteredTags.length) return notRegisteredTags[0];
+
+  // Find referenced tag in the tree
+  const controlledTag = flatTree.find(item => item.name === element.toname);
+
+  if (controlledTag === undefined) {
+    return errorBuilder.tagNotFound(model.name, "toname", element.toname);
+  }
+
+  if (controlledTags.validate(controlledTag.tagName).length > 0) {
+    return errorBuilder.tagUnsupported(model.name, "toname", controlledTag.tagName, controlledTags);
+  }
+
+  return null;
+};
+
+/**
+ * Validate other tag attributes other than name and toName
+ * @param {Object} child
+ * @param {import("mobx-state-tree").IModelType} model
+ * @param {string[]} fieldsToSkip
+ */
+const validateAttributes = (child, model, fieldsToSkip) => {
+  const result = [];
+  const properties = Object.keys(model.properties);
+
+  for (let key of properties) {
+    if (!child.hasOwnProperty(key)) continue;
```

Comment:
it should, but as it's mentioned in the description, there's no clear way of doing it because of `mayBeNull` usage all over the place. at least I haven't found one